schema.d.ts

```diff
@@ -299,6 +299,11 @@ export interface Attribute {
    * If the value is a string, it's the reason for the deprecation.
    */
   deprecated?: boolean|string;
+
+  /**
+   * Whether the attribute is read-only.
+   */
+  readonly?: boolean;
```

Comment:
What does it mean for an attribute to be read-only? afaiu, there really shouldn't be such a thing.

If there's a use for this, say to cover for the lack of CSS custom state, should we highlight that in the description?

Reply:
an element might observe changes in that attribute and reject values it finds icky, resetting to the previous known-good value, or removing the attribute.

Reply:
@justinfagnani Would this PR be clear to merge if it was only adding this to properties?  I can create a new PR if necessary, but it would be nice to get this added for properties to unblock the CEM analyzer support.  If @bennypowers or others feel strongly about usage with attributes, perhaps we can create a new issue for that discussion?

Reply:
If youre using @custom-elements-manifest/analyzer, you can already support this via a plugin

Reply:
I would go with this if it were for properties only to avoid semantics questions about "readonly" attributes.

Reply:
True, but they mentioned cem analyzer being blocked, which it isnt

Reply:
Thanks to both of you for the responses.  I am using that analyzer, and was able to find the plugin, so that does unblock me personally.  My use case for this particular feature is Angular Framework integration, and I need to ensure that readonly properties can't be bound as an `@Input()`.

I do still think it would be a good addition to the spec, so I'll try to get a PR up soon.  Enough people have asked and it encodes a real semantic of the language.  If it were added here, support could be added to the analyzer without a plugin, and other tools as well.

Reply:
Id be okay with adding `readonly` for fields, im also unsure about attributes. Perhaps we can indeed address that in a separate discussion.

@MikeMatusz If and when that change has landed in the schema, would you be willing to contribute your plugin logic to cem/a?